hacking/release-procedure.rst

```diff
@@ -0,0 +1,350 @@
+   cause problems like the website or tarball failing to build.  So we want to put this someplace
+   where those are not problems.
+
+   Additionally, process docs like this should go in without the docs team pre-review.  For process
+   docs, sharing of the process with other committers needs to take precedence otherwise the
+   committers will end up sharing the information outside of the documentation.  We desperately need
+   to document our step-by-step procedures so we need to remove barriers to entry for getting those
+   in.
+
+   In addition to this doc there's a few "templates" that I've linked to hat should come into the
+   repo too (template for release announcement email, etc)
+
+Release Manager Process
+=======================
+
+This document describes the step-by-step procedures for creating a release.
+
+
+Cherry-picking
+--------------
+
+* Create and populate a ``2.X.Y Blockers`` Github Project for each release (major or minor release).
+  Use it to track the blocker bugs for release.
+
+    * Use the `2.4.1 Blocker project <https://github.com/ansible/ansible/projects/11>`_ as a template.
+
+* Leave a comment in the PR that you cherry-pick from as to what release the change will show up in
+  example:
+
+   * ``Merged to devel and cherry-picked for the 2.4.0rc3 release.``
+   * Use cherry-pick -x to automatically add the SHA to the cherrypick::
+
+        git cherry-pick -x SHA
+
+* When cherry-picking to minor releases, add a changelog entry for every cherrypick.  Include a link
+  to the PR or bug.
+* For Beta periods, allow committers to use their judgement
+* For RC periods, ask mattclay/a github admin to lock down permissions on the branch so that only
+  the RM can cherry-pick
+
+
+Pre beta1
+---------
+
+* Coordinate with Project Manager about syncing open PRs from the Roadmap blocker.  Roadmap blocker
+  will have mainly features and higher level cards.
+
+   * Decide which open ones are to be punted vs finished.
+   * If to be finished one has a single PR, copy over.
+   * If to be finished one has multiple PRs, ask feature owner what the PRs are and move them all
+     over to the bug
+
+* Get the permissions you will need:
+
+   * Able to ssh and sudo on ``74.207.229.77`` (master mirror)
+   * Ops in ``#ansible``
+   * Access to http://jenkins.testing.ansible.com/
+   * Able to post to ``ansible-announce``, ``ansible-project``, and ``ansible-devel`` mailing lists
+
+
+.. _first_release:
+
+To make beta1 for the major release
+-----------------------------------
+
+1. Create stable-2.X and push
+2. Change version in stable-2.X:
+
+    a. Change devel to 2.X in :file:`docs/docsite/_themes/srtd/layout.html`
+
+3. Bump the **devel** version to 2.(X+1)
+
+    a. :file:`VERSION`
+    b. :file:`lib/ansible/release.py`
+    c. :file:`packaging/release/vars/versions.yml`
+
+        i. Determine a codename for 2.(X+1) (Led Zeppelin song name)
+
```

Comment:
Should we start a `devel` docs build here? (Killing any GitHub triggered jobs).

Reply:
Probably should make the docs after the release is cut.  Unless we want to ensure that docs build failing is a release blocker.